skrf/vectorFitting.py

```diff
@@ -2260,6 +2260,170 @@ def plot_passivation(self, ax: Axes = None) -> Axes:
+        >>> vf = skrf.VectorFitting(nw_3port)
+        >>> vf.vector_fit(n_poles_real=1, n_poles_cmplx=4)
+        >>> vf.write_spectre_subcircuit_new_s('/my3port_model.sp')
+
+        References
+        ----------
+        .. [1] G. Antonini, "SPICE Equivalent Circuits of Frequency-Domain Responses", IEEE Transactions on
+            Electromagnetic Compatibility, vol. 45, no. 3, pp. 502-512, August 2003,
+            doi: https://doi.org/10.1109/TEMC.2003.815528
+
+        .. [2] C. -C. Chou and J. E. Schutt-Ainé, "Equivalent Circuit Synthesis of Multiport S Parameters in
+            Pole–Residue Form," in IEEE Transactions on Components, Packaging and Manufacturing Technology,
+            vol. 11, no. 11, pp. 1971-1979, Nov. 2021, doi: 10.1109/TCPMT.2021.3115113
+
+        .. [3] Romano D, Antonini G, Grossner U, Kovačević-Badstübner I. Circuit synthesis techniques of
+            rational models of electromagnetic systems: A tutorial paper. Int J Numer Model. 2019
+            doi: https://doi.org/10.1002/jnm.2612
+
+        """
+
+        # List of subcircuits for the equivalent admittances
+        subcircuits = []
+
+        # Provides a unique subcircuit identifier (X1, X2, X3, ...)
+        def get_new_subckt_identifier():
+            subcircuits.append(f'X{len(subcircuits) + 1}')
+            return subcircuits[-1]
+
+        with open(file, 'w') as f:
+            # write title line
+            f.write('* EQUIVALENT CIRCUIT FOR VECTOR FITTED S-MATRIX\n')
+            f.write('* Created using scikit-rf vectorFitting.py\n\n')
+
+            # Define the complete equivalent circuit as a subcircuit with one input node per port
+            # those port nodes are labeled p1, p2, p3, ...
+            # all ports share a common node for ground reference (node 0)
+            str_input_nodes = " ".join(map(lambda x: f'p{x + 1}', range(self.network.nports)))
+
+            f.write(f'.SUBCKT {fitted_model_name} {str_input_nodes}\n')
+            for n in range(self.network.nports):
+                f.write(f'\n* Port network for port {n + 1}\n')
+
+                # Port reference impedance Z0
+                f.write(f'R{n + 1} p{n+1} a{n + 1} {np.real(self.network.z0[0, n])}\n')
+
+                # CCVS implementing the reflected wave b.
+                # Also used as current sensor to measure the input current
+                f.write(f'H_b_{n + 1} a{n + 1} 0 V_c_{n + 1} 1\n')
+
+                f.write(f'* Differential incident wave a sources for transfer from port {n + 1}\n')
+
+                # CCVS and VCVS driving the transfer admittances with 2*a*sqrt(Z0) = V + I*Z0
+                f.write(f'H_p_{n + 1} nt_p_{n + 1} nts_p_{n + 1} H_b_{n + 1} {np.real(self.network.z0[0, n])}\n')
+                f.write(f'E_p_{n + 1} nts_p_{n + 1} 0 p{n + 1} 0 1\n')
+
+                # VCVS driving the transfer admittances with -2*a*sqrt(Z0) = -V - I*Z0
+                f.write(f'E_n_{n + 1} 0 nt_n_{n + 1} nt_p_{n + 1} 0 1\n')
```

Comment:
Same applies here, I changed the E_p_{n+1} source's gain to get the exact definition of the a wave. Only the E_n_{n+1} source's gain is unchanged and 1 because it just inverts the sign of the a wave.